Source/WebCore/Modules/applepay-ams-ui/ApplePayAMSUIPaymentHandler.cpp

```diff
@@ -108,7 +108,7 @@ ApplePayAMSUIPaymentHandler::ApplePayAMSUIPaymentHandler(Document& document, con
 
 Document& ApplePayAMSUIPaymentHandler::document() const
 {
-    ASSERT(is<Document>(scriptExecutionContext()));
+    ASSERT(scriptExecutionContext());
```

Comment:
Should this ASSERT(dynamicDowncast<Document>(scriptExecutionContext())?

Reply:
The next line does `downcast<Document>(*scriptExecutionContext())` which already asserts that the scriptExecutionContext is a Document internally. As a result, this assertion was mostly redundant. The only extra check it did was the null check. Therefore, I converted the assertion to only do the null check.